pkg/statsd/handler_tags.go

```diff
@@ -28,25 +28,46 @@ func (th *TagHandler) EstimatedTags() int {
 	return th.estimatedTags
 }
 
-// DispatchMetric adds the tags from the TagHandler to the metric and passes it to the next stage in the pipeline
+// DispatchMetric adds the unique tags from the TagHandler to the metric and passes it to the next stage in the pipeline
 func (th *TagHandler) DispatchMetric(ctx context.Context, m *gostatsd.Metric) error {
 	if m.Hostname == "" {
 		m.Hostname = string(m.SourceIP)
 	}
-	m.Tags = append(m.Tags, th.tags...)
+	m.Tags = uniqueTags(m.Tags, th.tags)
 	return th.metrics.DispatchMetric(ctx, m)
 }
 
-// DispatchEvent adds the tags from the TagHandler to the event and passes it to the next stage in the pipeline
+// DispatchEvent adds the unique tags from the TagHandler to the event and passes it to the next stage in the pipeline
 func (th *TagHandler) DispatchEvent(ctx context.Context, e *gostatsd.Event) error {
 	if e.Hostname == "" {
 		e.Hostname = string(e.SourceIP)
 	}
-	e.Tags = append(e.Tags, th.tags...)
+	e.Tags = uniqueTags(e.Tags, th.tags)
 	return th.events.DispatchEvent(ctx, e)
 }
 
 // WaitForEvents waits for all event-dispatching goroutines to finish.
 func (th *TagHandler) WaitForEvents() {
 	th.events.WaitForEvents()
 }
+
+func uniqueTags(t1 gostatsd.Tags, t2 gostatsd.Tags) gostatsd.Tags {
+	tags := gostatsd.Tags{}
+	seen := map[string]bool{}
```

Comment:
Usually sets are modelled with `map[x]struct{}` - see e.g. https://www.davidkaya.sk/2017/12/10/sets-in-golang/
It is better for these reasons:
- it is idiomatic in Go. Others will immediately understand the point of code. If value is of type `struct{}` that means that value... has no value, only the presence/absence of it is valuable.
- More memory efficient.